File: easy/python/1009.py
class Solution:
    def bitwiseComplement(self, n: int) -> int:
        binary_string_n = format(n, "b")
        array = []

        for num in binary_string_n:
            if num == "0":
                array.append("1")
            else:
                array.append("0")
        return int("".join(array), 2)

File: easy/ts/1009.ts
// https://leetcode.com/problems/complement-of-base-10-integer/
// Time: O(n)
// Space: O(n)
function bitwiseComplement(n: number): number {
  let binaryN = n.toString(2);
  let newBinaryNumArr: string[] = [];
  for (let i = 0; i < binaryN.length; i++) {
    if (binaryN[i] === "0") {
      newBinaryNumArr.push("1");
    } else {
      newBinaryNumArr.push("0");
    }
  }
  return parseInt(newBinaryNumArr.join(""), 2);
}
